### ISSUE TITLE
Spurious "multiple rules generate" error

### ISSUE THREAD

**Issue by anarazel** [ninja-build/ninja#2401]
Hi,

Starting with #2358 building postgres with meson and ninja from git triggers:
```
build.ninja:5628: multiple rules generate src/backend
```

The line number looks spurious - it's an empty line and neither of the neighboring lines actually generate "src/backend".

Repro:
```
mkdir /tmp/repro
cd  /tmp/repro

git clone https://github.com/ninja-build/ninja.git
git clone https://github.com/mesonbuild/meson.git
git clone https://github.com/postgres/postgres.git

cd ninja
git co 4d38849501ca09c2dcd9e778b59f9a4ec32180dd
./configure.py --bootstrap

cd ../postgres
../meson/meson.py setup build

../ninja/ninja -C build/ -d stats
ninja: Entering directory `build/'
ninja: error: build.ninja:5628: multiple rules generate src/backend
```

CC: @digit-google 

Greetings,

Andres

**Comment by digit-google** [comment]
Thank you for the reproduction case! I could take a look at the generated build.ninja file.
Good news, this should be fixed once PR #2398 is submitted.

The error comes from the line above that defines a build statement for the following target: `src/backend/postgres_lib.a.p/.._timezone_pgtz.c.o`. Due to a bug introduced in #2358, and fixed in #2398, this incorrectly canonicalized the path to `src/backend`.

**Comment by anarazel** [comment]
Thanks, can confirm that this fixed the issue for me!
